### ISSUE TITLE
Fitting bulkheads with the -100% CPU reduction results in spurious warning about needing more CPU

### ISSUE THREAD

**Issue by ryllae** [zhyrohaad/AlasiyaEvE#36]
Kronos added a role bonus of "100% reduction in CPU requirements for Reinforced Bulkheads" to all freighters, but it shouldn't be on the ship yet.  So I decided to fit some and see what happens.   You get the warning that you don't have the necessary 40 CPU (out of 1.25!), but yet you can right click the bulkheads and enable them.  If you undock and disable them, you can't reenable them, as they still require 40 CPU in space.
![Capture31](https://user-images.githubusercontent.com/51723245/63743923-6bf44980-c863-11e9-952c-ab83a1a2a032.PNG)

**Comment by ryllae** [comment]
And even if you want to keep the Kronos features, the fitting error message should check the ship for -CPU% bonuses first.

**Comment by zhyrohaad** [comment]
The effects are taken from the Rhea expansion.  
I am aware of some slight discrepancies in ship resources between docked and space.  Not sure how to fix that.

**Comment by ryllae** [comment]
Ok, so the freighter HAS the Kronos effect, but still has the description and stats from before the rework - that entirely https://github.com/zhyrohaad/AlasiyaEvE/issues/35   I'll update this bug to just address the fitting issue, which is separate.

**Comment by ryllae** [comment]
Ergo: "when fitting, the server should take into account the ship's CPU reduction before giving an error message".

The difference between doing this in drydock and space is probably relevant somewhere in the fitting versus onlining code, but both should (ideally) take this bonus into account properly.

**Comment by zhyrohaad** [comment]
i cant affect how the client does anything.  the best i can do is try to make the server match.

however, when docked, the server is not consulted for fitting, so that's all clientside.

**Comment by ryllae** [comment]
So this client isn't ready for Freighters to have low slots.  Guess we'll just have to bump the server up to Rhea. ;)

**Comment by THUNDERGROOVE** [comment]
Well, correcting the data set for freighters to match what they should be for crucible is a definite possibility.  Just a matter of replacing all of the related dogma shit for them.  All of the data is in the client already so just a matter of writing a script.

**Comment by zhyrohaad** [comment]
the data has been replaced, so it would have to be taken from a base install that HASNT connected to alasiya.

**Comment by THUNDERGROOVE** [comment]
Yep, fully aware.  Is this something you think is worth pursuing?

**Comment by zhyrohaad** [comment]
in all honesty, im not sure just how much changed between Crucible and Rhea.
however, as this code issue shows, this could pose a problem.
this seems as if the client is still trying to use Crucible data, but is becoming confused, which tells me there is more to the data than just the updates i've made.
is that what you were referring to by `All of the data is in the client already....`?  

if that's the case, and there is STILL data that isnt matching, then yes, this is something definitely worth pursuing, at very least to cut down bugs and things i cannot control (i.e.  fit window when docked).

actually, either way i think this is worth pursuing.

updating the client with this change (and any others we see fit or need to do) can be either launcher or bulkdata.....as im sure you know, with bulkdata preferable, as it cant be disabled.

**Comment by ryllae** [comment]
I think you guys are actually arguing #35, as this *specific* bug is to the fact that the client doesn't handle the CPU checks correctly - Crucible client never expected the freighter to even have low slots or take advantage of said bonus.

**Comment by zhyrohaad** [comment]
thats right.
what we are saying is this.....
because the client isnt recognizing the data we're using (from Rhea), we will have to revert back to crucible data for things like this (as they come up).
what we're saying is the WHY part of this particular bug, and how to fix it, which also ties into #35, and may be others we havent found yet.
there are some things we have changed that the client is ok with, and checks correctly.
this is not one of them.

basically, this bug, and others like it, will have to wait until we can get 'good' data and update client.
i dont know how long that will take.

**Comment by zhyrohaad** [comment]
found orig data
fixed and updated

**Comment by ryllae** [comment]
since you can no longer equip anything on a freighter re: #35, this is moot unless we move the bulkdata ahead again.

**Comment by zhyrohaad** [comment]
Which we will not in this version.
